lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -85,6 +90,39 @@ export class PublicClientApplication extends ClientApplication implements IPubli
         if (!account) {
             throw BrowserAuthError.createNoAccountError();
         }
+        const thumbprint: RequestThumbprint = {
+            clientId: this.config.auth.clientId,
+            authority: request.authority || "",
+            scopes: request.scopes,
+            homeAccountIdentifier: account.homeAccountId
+        };
+        const silentRequestKey = JSON.stringify(thumbprint);
+        let response = this.activeSilentTokenRequests.get(silentRequestKey);
+        if (typeof response === "undefined") {
+            this.logger.verbose("acquireTokenSilent called for the first time, storing active request", request.correlationId);
+            response = this.acquireTokenSilentAsync(request, account)
+                .then((result) => {
+                    this.activeSilentTokenRequests.delete(silentRequestKey);
```

Comment:
Should we cache and return responses for requests with the same thumbprint for longer than the duration of the request? It might require some thought around lifecycle for cached responses but maybe something to consider. Might be too big for this PR though.

Reply:
Isn't that what local/session storage is for?

Reply:
not sure what you mean

Reply:
Why would we need to hold on to the `activeSilentTokenRequests` object longer than the duration of the request? After the request is complete any subsequent calls to ATS should pull from local/session storage

Reply:
Was just thinking that caching the actual response object would be better than doing the entire lookup again, but it's probably an over-optimization here.